### ISSUE TITLE
transfer learning from AMOS dataset to KiTS dataset

### ISSUE THREAD

**Issue by joeyzyz** [JiYuanFeng/AMOS#3]
Great code and very impressive paper. Thank you very much!

I am writing to consult problem with transfer learning from AMOS dataset to KiTS dataset.

In the table 3 of the paper, you pre-trained with AMOS, then apply it on KiTS dataset, finally, you got **88.33** dice score, I found it is very interesting, so I tried the same thing, but only got **65.54** dice score. I don't know what is went wrong, the following is the potential problem:

1. The two dataset has different axis ordering, AMOS dataset is (z,y,x), while KiTS is (x,y,z). Did you doing anything to preprocess it or you just ignore it?
2. The two dataset has different voxel spacing, how did you preprocess them to become same target spacing?

Hope to get some response from you. Thank you~

**Comment by JiYuanFeng** [comment]
Hi!, thanks for your interest in our work. For the question:
1. Yes, there have different axis ordering, please modify the related experimental configuration when testing on Kits, it should be a file like "xxx.pkl" under the experiment folder.
2. No, although they have different spacings, the predictions made by the model trained on AMOS will re-sample to the original spacing, and we can compare them with ground truth directly.

**Comment by joeyzyz** [comment]
> Hi!, thanks for your interest in our work. For the question:
> 
> 1. Yes, there have different axis ordering, please modify the related experimental configuration when testing on Kits, it should be a file like "xxx.pkl" under the experiment folder.


Thanks for your reply.  @JiYuanFeng 

which part of the "xxx.pkl" did you modify? That is, which key did you change in the pickle file, and how did you change it?

**Comment by JiYuanFeng** [comment]
Hi, I checked my previous code and you should modify the “plans.pkl” in your model folder. For the keywords, it should be 'transpose_forward' and ''transpose_backward''.  You can try to modify to ['transpose_forward']=[2,0,1].

**Comment by joeyzyz** [comment]
> Hi, I checked my previous code and you should modify the “plans.pkl” in your model folder. For the keywords, it should be 'transpose_forward' and ''transpose_backward''. You can try to modify to ['transpose_forward']=[2,0,1].

Hi @JiYuanFeng , Thanks for your reply.

After I changed **transpose_forward** and **transpose_backward** to the following:
`
transpose_forward = [2, 0, 1] ,
transpose_backward = [1, 2, 0]
`
the dice score just stay at about **65** and not willing to improve any more.

The potential reason I guess is that I only trained with **100** epochs, I will evaluate again when I train for **1000** epochs.

Meanwhile, I have some question under transfer learning with fine-tuning:

1. In table 3 of the paper, you pre-trained with AMOS dataset and **direct predicted** on KiTS dataset, while in table 6 of the paper, you pre-trained with AMOS dataset and **fine-tuning** on KiTS dataset. The control group for both experiments was the same, which is training from scratch, but table 3 got **97.47** dice score in control group, and table 6 got only **88.52** dice score. This brings inconsistency in same experiment, I guess maybe the two control group training for different epochs, what if you trained enough epochs (say 1000 epochs), would the conclusion of table 6 change? There maybe two potential result: 
- pre-training on amos and fine-tuning on kits still improve the final dice score
- pre-training on amos and fine-tuning on kits only accelerate the convergence in the early stage, and the finally dice score is still unimproved. 
Which one do you think maybe incorrect?

2. In fine-tuning setting, preprocessing should be different from that of direct predicting in table 3, since in this setting, we need to training with **two** different datasets. How did you preprocess the two dataset to make them have same axis ordering, intensity normalization,and the target spacing?

I have a lot of questions, I try to keep them as concise as possible, really hope to discuss with you, and hope it will not take up too much of your time, thank you very much~

**Comment by JiYuanFeng** [comment]
Hi, @joeyzyz, thank you for your question. 

For the "transpose_forward and transpose_backward",  I will check again after my paper submission. Generally, there are two places that can set these two keys on the overall nnunet pipeline.  

For other questions: 
1. Lets me describe the difference between the two experiments. For the experiments in table 3, since the Kits provide the tumor and kidney annotation, and AMOS only provides the kidney label (that is, even if there are some lesions on the kidney, we still label them "kidney"), for direct comparison, we merge two classes of the kits (i.g. tumor, kidney -> "kidney") and compute metric with the prediction from the model trained on AMOS (only "kidney" classes), so it has a high score (97.43 for only "kidney"). While in Table 6, the metric is averaged by two categorical metrics (tumor, kidney), and it achieves a much lower score. Overall, **pre-training on AMOS and fine-tuning on kits (compared to training from scratch) will improve the final dice score (in both classes)**.
2. For the fine-tuning setting,  we did not consider aligning the same axis ordering, intensity normalization, and the target spacing between two datasets (source: AMOS, target: MSD, kits...), we only train different architectures with various down sample strides (see below) using default training configuration (amos), and then directly use the trained weight for the initialization of new tasks (using the new configuration of target dataset). Overall, we only align the unet architecture between source and target datasets.
<img width="230" alt="image" src="https://user-images.githubusercontent.com/20653249/191158468-3c225b3a-127c-4a1a-b7f3-3dd3e03300c2.png">

**Comment by joeyzyz** [comment]
Very useful answer, thank you very much~
